compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -479,8 +481,10 @@ pub enum ValidationErrorKind<'tcx> {
         value: String,
     },
     InvalidMetaWrongTrait {
-        expected_trait: &'tcx ty::List<ty::PolyExistentialPredicate<'tcx>>,
-        vtable_trait: Option<ty::PolyExistentialTraitRef<'tcx>>,
+        /// The vtable that was actually allocated into global memory.
+        allocated_vtable: &'tcx ty::List<ty::PolyExistentialPredicate<'tcx>>,
+        /// The vtable that was expected at the point in MIR that it was accessed.
+        expected_vtable: &'tcx ty::List<ty::PolyExistentialPredicate<'tcx>>,
```

Comment:
"allocated" sounds odd here. IMO the previous distinction makes more sense -- the "expected" info comes from the type, and that is compared with the actual info stored in the vtable pointer. There is no "expected vtable", there's just an expected list of predicates. IOW, `List<ty::PolyExistentialPredicate<'tcx>>` isn't a vtable, only `GlobalAloc::Vtable` is a vtable.

Reply:
Well, I mostly had a problem with calling it a "trait". It's now not a trait, it's ... poly_existential_predicates which is a mouthful. I understand vtable is a misnomer here, but do you have a better suggestion for a name?

Reply:
I think "allocated" also here makes total sense, it's the vtable that was allocated in the first place. "expected" here is what we expected the vtable to end *up* as; that's the unsoundness.

It's like calling the type of an allocation its "allocated type" and the type you end up making an invalid read on some place as its "expected type". Again, I'm totally happy to rename this, but I'd rather not call it "trait" when it's no longer an (existential) trait.

Reply:
Maybe vtable → dyn type? After all this list is exactly all the things that go into a `dyn Trait` type.

Then I'd use `expected_dyn_type` and `vtable_dyn_type`.

Reply:
Ya that's fine 👍